tools/model-compiler/src/main/java/io/spine/tools/gradle/compiler/SpineCheckerPlugin.java

```diff
@@ -0,0 +1,350 @@
+        addArgsToJavaCompile(project, "-processorpath", preprocessorConfig.getAsPath());
+    }
+
+    /**
+     * Converts the {@link #configureCheckSeverity(Project)} method to the Gradle {@link Action}.
+     */
+    private Action<Gradle> configureSeverityAction(Project project) {
+        return gradle -> configureCheckSeverity(project);
+    }
+
+    /**
+     * Adds command line flags necessary to configure Spine check severities to all
+     * {@code JavaCompile} tasks of the project.
+     */
+    private void configureCheckSeverity(Project project) {
+        if (!hasErrorPronePlugin(project)) {
+            log().debug("Cannot configure Spine checks severity as Error Prone plugin is not " +
+                                "applied to the project {}.", project.getName());
+            return;
+        }
+        Severity defaultSeverity = Extension.getSpineCheckSeverity(project);
+        configureUseValidatingBuilder(project, defaultSeverity);
+    }
+
+    /**
+     * Configures the "UseValidatingBuilder" check severity for all {@code JavaCompile} tasks of
+     * the project.
+     *
+     * <p>Uses default severity set in the {@code modelCompiler} extension if set and not
+     * overridden by the more specific {@code spineChecker} extension.
+     */
+    @SuppressWarnings("ConstantConditions") // Checking nullable argument for null.
+    private void
+    configureUseValidatingBuilder(Project project, @Nullable Severity defaultSeverity) {
+        Severity severity = getUseValidatingBuilder(project);
+        if (severity == null) {
+            if (defaultSeverity == null) {
+                return;
+            } else {
+                severity = defaultSeverity;
+            }
+        }
+        log().debug("Setting UseValidatingBuilder checker severity to {} for the project {}",
+                    severity.name(), project.getName());
+        String severityArg = "-Xep:UseValidatingBuilder:" + severity.name();
+        addArgsToJavaCompile(project, severityArg);
+    }
+
+    /**
+     * Adds specified arguments to all {@code JavaCompile} tasks of the project.
+     */
+    private static void addArgsToJavaCompile(Project project, String... args) {
+        TaskContainer tasks = project.getTasks();
+        TaskCollection<JavaCompile> javaCompileTasks = tasks.withType(JavaCompile.class);
+        for (JavaCompile task : javaCompileTasks) {
+            CompileOptions taskOptions = task.getOptions();
+            List<String> compilerArgs = taskOptions.getCompilerArgs();
+            compilerArgs.addAll(Arrays.asList(args));
+        }
+    }
+
+    /**
+     * Checks if the project has Error Prone plugin applied.
+     */
+    @VisibleForTesting
+    protected boolean hasErrorPronePlugin(Project project) {
+        PluginContainer appliedPlugins = project.getPlugins();
+        boolean hasPlugin = appliedPlugins.hasPlugin(ERROR_PRONE_PLUGIN_ID);
+        return hasPlugin;
+    }
```

Comment:
Can we please try to keep the plugin class as small as possible by moving the code into other helper-filter-etc classes. See how it's done for `ModelCompilerPlugin`.